ddp-workspace/projects/ddp-atcp/src/app/components/activityForm/app-atcp-activity-base.component.ts

```diff
@@ -254,6 +257,15 @@ export class AtcpActivityBaseComponent extends ActivityComponent implements OnIn
           }
         })
     );
+    this.anchor.addNew(
+      combineLatest([this.languageService.onLanguageChange(), this.languageService.getParticipantGuidBeforeLanguageChange()])
+      .pipe(delay(2000))
+      .subscribe(
+          ([, participantGuid]) => {
+            !!participantGuid && this.session.setParticipant(participantGuid);
+            this.languageService.notifyOfProfileLanguageUpdate();
```

Comment:
the issue occurs due to the fact that the translations are getting from the backend instead of using ngx-translate library in angular. In order to do so we have to trigger the notification through the service above `( this.languageService.notifyOfProfileLanguageUpdate();)` whenever the language changed.
However, to proceed with that it was necesssary to retrieve the participantGuid and set it to the current session (That was the tricky part) since I was getting an error trying to call the backend endpoint with the wrong participantGuid.